R/validation.R

```diff
@@ -253,3 +200,24 @@ validate_estimator <- function(estimator, estimator_override = NULL) {
   }
 
 }
+
+validate_case_weights <- function(case_weights, size) {
+  if (is.null(case_weights)) {
+    return(invisible())
+  }
+
+  if (!is.integer(case_weights) && !is.double(case_weights)) {
+    abort("`case_weights` must be an integer or double vector.")
+  }
```

Comment:
Was I doing any validation of the `case_weights` type before hand? I think this technically works if people provide hardhat's case weight types, but ideally `is.integer()` would actually return `FALSE` for those. It happens to return `TRUE` because those are the storage types of the vectors.

I think I was previously just letting `yardstick_table()` -> `hardhat::weighted_table()` handle the validation of the `case_weights` because it uses `vec_cast()`